File: acoplamiento/problema3/editor.py
class Editor:
    def check_spelling(self, text):
        errors = self.check(text)
        if errors:
            self.display_errors(errors)
        else:
            self.display_success_message()

    def display_errors(self, errors):
        for error in errors:
            print("ERROR: {}".format(error))
        return None

    def display_success_message(self):
        print("No errors found!")
        return None

    def check(self, text):
        words = text.split()
        errors = []
        for word in words:
            if word not in ['foo', 'bar']:
                errors.append(word)
        return errors

    def runEditor(self):
        print("Running editor...")
        print("Enter text:")
        text = input()
        self.check_spelling(text)

if __name__ == "__main__":
    editor = Editor()
    editor.runEditor()

File: acoplamiento/problema3/src/editor/Editor.java
package editor;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;

public class Editor {
	
	public void checkSpelling(String text) {
		ArrayList<String> errors = this.check(text);
		if(errors != null)
			this.displayErrors(errors);
		else
			this.displaySuccessMessage();
	}
	
	public void displayErrors(ArrayList<String> errors) {
		for(int i=0; i<errors.size(); i++) {
			System.out.println("ERROR: " + errors.get(i));
		}
	}
	
	public Object displaySuccessMessage() {
		System.out.println("No errors found!");
		return null;
	}
	
	public ArrayList<String> check(String text) {
		String[] words = text.split("");
		ArrayList<String> errors = new ArrayList<String>();
		for(int i=0; i< words.length; i++) {
			String[] realWords = {"foo", "bar"};
			for(int j=0; j<realWords.length; j++) {
				if(!words[i].equalsIgnoreCase(realWords[j])) {
					errors.add(words[i]);
				}
			}
		}
		return errors;
	}
	
	public void runEditor() {
		System.out.println("Running editor...");
		System.out.println("Enter text:");
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String text;
		try {
			text = br.readLine();
			this.checkSpelling(text);
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
	
	public static void main(String[] args) {
		Editor editor = new Editor();
		editor.runEditor();
	}
}
